pipelines/train_model.py

```diff
@@ -0,0 +1,136 @@
+import polars as pl
+from prefect import flow, task
+from prefect_ray import RayTaskRunner
+
+from pipelines import lstm, transformations
+from pipelines.loaders import load_dataframe
+from pipelines.types import ColumnSubset, TimeWindow
+
+
+@task
+def valid_size(data, window: TimeWindow):
+    return data.shape[0] >= window.length
+
+
+@task
+def train_val_test_split(
+    data: pl.DataFrame, splits: tuple[int, int, int], preserve_order: bool = True
+):
+    if not preserve_order:
+        data = data.shuffle()
```

Comment:
For my own edification: when would we not want to preserve order for a timeseries prediction?

Reply:
in general there is no reason you wouldn't, I think in the original code you had a `preserve_order` boolean that was more or less hardcoded to `True`.

The way that the data was getting split up, honeslty I don't know that it matters a ton in the sense that if you slice it so each observation contains the ordered data, then any individual datapoint will be ordered but the ordering of those won't matter.

To be fair, this is one of those places where I just don't know TF as well as pytorch, but looking at the docs that is my understanding.

For clarity, we can just straight up remove the option to shuffle.

Reply:
Definitely fine keeping it - it makes sense with your explaination.